src/spec/test/DesignPatternsTest.groovy

```diff
@@ -204,7 +204,7 @@ class DesignPatternsTest extends CompilableTestSupport {
             // tag::adapter_inheritance_usage[]
             def hole = new RoundHole(radius: 4.0)
             (4..7).each { w ->
-                def peg = new SquarePegAdapter(peg: new SquarePeg(width: w))
+                def peg = new SquarePegAdapter(width: w)
```

Comment:
Nice catch. This is the  weakness of this example. It uses println statements instead of asserts so we just used shouldCompile rather than assertScript, otherwise we would have caught this.